src/storage/mod.rs

```diff
@@ -61,60 +66,60 @@ pub enum Command {
         ctx: Context,
         key: Key,
         start_ts: u64,
-        callback: Callback<Option<Value>>,
+        callback: Option<Callback<Option<Value>>>,
```

Comment:
When do we have None callback? I see you use take().unwrap() for nearly all callback handling.